### PR TITLE
Manual - note on calling conventions on Win64

### PR DESCRIPTION
**Pull request by Petr-Hlavenka**
Clarification based on discussion in mailing list https://discourse.julialang.org/t/callback-to-winapi/1381

### CODE REVIEW

File: doc/src/manual/calling-c-and-fortran-code.md [hunk 1]
```diff
@@ -820,6 +820,8 @@ hn = Array{UInt8}(256)
 err = ccall(:gethostname, stdcall, Int32, (Ptr{UInt8}, UInt32), hn, length(hn))
 ```
 
+Note that for 64-bit Windows programs the calling conventions have been unified and even for
```
Comment (tkelman) [inline doc/src/manual/calling-c-and-fortran-code.md]: remove trailing whitespace

**Comment by phlavenk** [comment]
Removed.

**Comment by vtjnash** [comment]
This seems sort of misleading to me, since Win64 has at least 2 calling conventions officially defined. Also, for compatibility with Win32, it is also convenient to specify the calling convention, even though at the ABI level it doesn't change anything.

**Comment by phlavenk** [comment]
Based on [MSDN](https://msdn.microsoft.com/en-us/library/ms235286.aspx?f=255&MSPPError=-2147217396), there is only one calling convention. I was confused when writing a WINAPI ccall with a callback function and i couldn't find `cfunction` where i could specify calling convention. @ihnorton clarified the situation for me that on Win64 I don't have to worry about as Win64 only uses one calling convention.

On the other hand, the ability to call into WINAPI is pretty important (for win32 users without the full power of bash, etc.) and may deserve some notice in manual.
I can delete it from calling convention and move a remark to the callback section, that only C-style callback is available on Win32, but on Win64 you are save to proceed.

**Comment by ihnorton** [comment]
I would suggest:
- add `(no-op on 64-bit Windows)` after sentence `Other supported conventions are: stdcall, cdecl, fastcall, and thiscall.`
- add note to `cfunction` description (and help) that `cfunction` only supports platform-default CC.

(related: #5613)

**Comment by phlavenk** [comment]
Agree. I'll update the PR.

**Comment by phlavenk** [comment]
I don't think it is necessary to do changes in the help string, only changed the manual.

**Comment by ihnorton** [comment]
LGTM

File: doc/src/manual/calling-c-and-fortran-code.md [hunk 2]
```diff
@@ -147,12 +147,16 @@ typedef returntype (*functiontype)(argumenttype,...)
...
 The function [`cfunction()`](@ref) generates the C-compatible function pointer for a call to a
-Julia library function. Arguments to [`cfunction()`](@ref) are as follows:
+Julia function. Arguments to [`cfunction()`](@ref) are as follows:
 
 1. A Julia Function
 2. Return type
 3. A tuple of input types
 
+Only platform-default C calling convention is supported. `cfunction`-generated pointers cannot
+be used in calls where WINAPI expects `stdcall` function on 32-bit windows, but can be used on WIN64
```
Comment (tkelman) [inline doc/src/manual/calling-c-and-fortran-code.md]: trailing whitespace

**Comment by phlavenk** [comment]
White space removed. Is there an easy way to check for the trailing whitespaces?

**Comment by tkelman** [comment]
`make check-whitespace`

**Pull request merged**